pkg/pillar/agentlog/loglevel.go

```diff
@@ -39,16 +39,22 @@ func getLogLevelImpl(sub *pubsub.Subscription, agentName string,
 		log.Infof("GetLogLevel failed %s\n", err)
 		return "", false
 	}
-	gc := cast.CastGlobalConfig(m)
-	// Do we have an entry for this agent?
-	as, ok := gc.AgentSettings[agentName]
-	if ok && as.LogLevel != "" {
-		return as.LogLevel, true
-	}
-	// Do we have a default value?
-	if allowDefault && gc.DefaultLogLevel != "" {
-		return gc.DefaultLogLevel, true
+	gc := cast.CastConfigItemValueMap(m)
+
+	if _, ok1 := gc.AgentSettings[agentName]; ok1 {
+		as, ok := gc.AgentSettings[agentName]["loglevel"]
+		if ok && as.StringValue != "" {
+			return as.Key, true
+		} else if ok && as.StringValue != "" && allowDefault {
+			// Return default loglevel (global)
+			return "global", true
+		} else {
+			log.Infof("Failed to find loglevel settings for %s", agentName)
+		}
+	} else {
+		log.Infof("Failed to find per agent settings for %s", agentName)
```

Comment:
This is not correct.. If we don't find Agent Level setting - we fall back to global setting.. See above comments..